lib/sqlalchemy/engine/result.py

```diff
@@ -177,6 +179,29 @@ def _row_as_tuple_getter(
         indexes = self._indexes_for_keys(keys)
         return tuplegetter(*indexes)
 
+    def _make_key_to_index(
+        self, keymap: Mapping[_KeyType, Sequence[Any]], index: int
+    ) -> Mapping[_KeyType, int]:
+        return {
+            key: rec[index]
+            for key, rec in keymap.items()
+            if rec[index] is not None
+        }
+
+    def _key_not_found(self, key: Any, attr_error: bool) -> NoReturn:
+        if key in self._keymap:
+            # the index must be none in this case
+            if attr_error:
+                try:
+                    self._key_fallback(key, KeyError(key))
```

Comment:
note to self: remove keyerror from the key_fallback, since it does not make much sense now